src/Quartz/Impl/JobDetailImpl.cs

```diff
@@ -236,7 +300,7 @@ public JobKey Key
         public string? Description
         {
             get => description;
-            set => description = value;
+            private set => description = value;
```

Comment:
Personally, I'd prefer to remove all setters (and make them - and the type itself - sealed).
But I know this could be considered a major breaking change.

Reply:
Well if this is v4 work... we can break as much as we want. We can always re-open sealed classes with good reasoning behind the request.